lib/Service/StackService.php

```diff
@@ -362,4 +368,93 @@ public function reorder($id, $order) {
 
 		return $result;
 	}
+
+	/**
+	 * @param $id
+	 * @param $boardId
+	 * @return Stack
+	 * @throws StatusException
+	 * @throws BadRequestException
+	 */
+	public function clone($id, $boardId) {
+		if (is_numeric($id) === false) {
+			throw new BadRequestException('stack id must be a number');
+		}
+
+		$this->permissionService->checkPermission(null, $boardId, Acl::PERMISSION_MANAGE);
```

Comment:
Please check if the user has read permission on the stack as well

Reply:
@jakobroehrl This is not checked yet.

Reply:
Nope, as mentioned we also need to check if the original ***stack can be read**

Otherwise a user could basically clone any stack from other users boards to their own ones.